### PR TITLE
Bump com.google.api-client:google-api-client from 1.34.0 to 2.2.0 in /plugins/repository-gcs

### PR DESCRIPTION
**Pull request by Xtansia**
### Description
Updates the com.google.api-client:google-api-client and the com.google.apis:google-api-services-storage dependencies.

### Related Issues
Supersedes #8008 

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: plugins/repository-gcs/build.gradle [hunk 1]
```diff
@@ -52,33 +52,43 @@ versions << [
...
+
+  runtimeOnly "com.google.guava:guava:${versions.guava}"
+  api 'com.google.guava:failureaccess:1.0.1'
+
+  api 'com.google.http-client:google-http-client:1.43.2'
   api 'com.google.http-client:google-http-client-appengine:1.43.2'
-  api 'com.google.http-client:google-http-client-jackson2:1.42.3'
   api 'com.google.http-client:google-http-client-gson:1.41.4'
-  api 'com.google.api:gax-httpjson:0.103.1'
+  api 'com.google.http-client:google-http-client-jackson2:1.42.3'
```
Comment (reta) [inline plugins/repository-gcs/build.gradle:81]:
```suggestion
  api 'com.google.http-client:google-http-client-jackson2:1.43.2'
```

File: plugins/repository-gcs/build.gradle [hunk 2]
```diff
@@ -52,33 +52,43 @@ versions << [
...
+
+  api 'com.google.oauth-client:google-oauth-client:1.33.3'
```
Comment (reta) [inline plugins/repository-gcs/build.gradle:83]:
```suggestion
  api 'com.google.oauth-client:google-oauth-client:1.34.1'
```

**Comment by Xtansia** [comment]
@reta Have updated those additional packages

**Comment by Xtansia** [comment]
The com.google.cloud deps can also be updated, but require more work due to changes in behaviour of client, so would do them separately

**Review by reta — APPROVED** [review]

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
@Xtansia needs a manual backport, please

**Pull request merged**